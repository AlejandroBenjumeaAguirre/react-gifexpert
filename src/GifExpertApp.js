import React, { useState } from 'react';
import CategoryAdd from './components/CategoryAdd'; 
import GifGrid from './components/GifGrid';


const GifExpertApp = () => {

    //const categories = ['Dragon Ball', 'Los caballeros del Zodiaco', 'Death Note']

    const [categories, setCategories] = useState(['Death Note']);

    /* const handleAdd = () => {
        const nuevo = 'Pokemon';

        setCategories((c) => [...c, nuevo]);
    } */

    return (
        <>
            <h2>GifExpertApp</h2>
            <CategoryAdd setCategories={setCategories} />
            <hr></hr>

            {/* <button onClick={handleAdd}>Agregar</button> */}

            <ul>
                { 
                    categories.map( category => 
                        <GifGrid key={category} category={ category } />
                    )
                }
            </ul>
        </>
    )
}

export default GifExpertApp;