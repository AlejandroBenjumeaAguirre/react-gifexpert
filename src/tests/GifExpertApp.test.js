import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Prueba en <GifExpertApp />', () => {
    
    test('Debe mostrarse correctamente', () => {
        
        const wrapper = shallow(<GifExpertApp />)

        expect( wrapper ).toMatchSnapshot();

    });
    
});