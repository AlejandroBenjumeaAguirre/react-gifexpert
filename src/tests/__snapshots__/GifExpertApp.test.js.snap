// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifExpertApp /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="Death Note"
      key="Death Note"
    />
  </ul>
</Fragment>
`;
